### PR TITLE
fix(auth): enhance the URL check

### PR DESCRIPTION
**Pull request by zyxxoo**

### CODE REVIEW

File: hugegraph-server/hugegraph-api/pom.xml [hunk 1]
```diff
@@ -164,6 +164,12 @@
             <artifactId>arthas-packaging</artifactId>
             <version>${arthas.version}</version>
         </dependency>
+      <dependency>
+        <groupId>org.gridkit.jvmtool</groupId>
+        <artifactId>sjk-core</artifactId>
+        <version>0.22</version>
+        <scope>compile</scope>
+      </dependency>
```
Comment (imbajin) [inline hugegraph-server/hugegraph-api/pom.xml:172]: TODO: we need add license info for them

File: hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AuthenticationFilter.java [hunk 2]
```diff
@@ -314,9 +316,8 @@ public boolean equals(Object obj) {
 
     public static boolean isWhiteAPI(ContainerRequestContext context) {
         String path = context.getUriInfo().getPath();
-
         for (String whiteApi : WHITE_API_LIST) {
-            if (path.endsWith(whiteApi)) {
+            if (MATCHER.match(whiteApi, path)) {
```
Comment (imbajin) [inline hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/filter/AuthenticationFilter.java:320]: TODO: add some tests for it

File: hugegraph-server/hugegraph-api/pom.xml (hunk 1 above)
Comment (zyxxoo) [reply hugegraph-server/hugegraph-api/pom.xml:172]: the dependency has existed

**Comment by SunnyBoy-WYH** [comment]
seems this pr will  also solve https://github.com/apache/incubator-hugegraph/issues/2176

**Review by javeme — APPROVED** [review]

**Review by imbajin — DISMISSED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -51,11 +51,12 @@ For example:
 
 <!-- DO NOT REMOVE THIS SECTION. CHECK THE PROPER BOX ONLY. -->
 
-- [ ]  Nope
-- [ ]  Dependencies (add/update license info) <!-- Don't forget to add/update the info in "LICENSE" & "NOTICE" files (both in root & dist module) -->
+- [ ]  Dependencies ([add/update license](https://hugegraph.apache.org/docs/contribution-guidelines/contribute/#321-check-licenses) info & [regenerate_dependencies](../hugegraph-server/hugegraph-dist/scripts/dependency/regenerate_known_dependencies.sh)) <!-- Don't forget to add/update the info in "LICENSE" & "NOTICE" files (both in root & dist module) -->
```
Comment (imbajin) [inline .github/PULL_REQUEST_TEMPLATE.md:54]: @simon824 update here

**Review by imbajin — APPROVED** [review]

**Review by VGalaxies — APPROVED** [review]

**Pull request merged**